// react-grab/dist/index.cjs
'use strict';

var chunk6ZPKD3CA_cjs = require('./chunk-6ZPKD3CA.cjs');

/**
 * @license MIT
 *
 * Copyright (c) 2025 Aiden Bai
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

// src/index.ts
var globalApi = null;
var getGlobalApi = () => {
  if (typeof window === "undefined") return globalApi;
  return window.__REACT_GRAB__ ?? globalApi ?? null;
};
var setGlobalApi = (api) => {
  globalApi = api;
  if (typeof window !== "undefined") {
    if (api) {
      window.__REACT_GRAB__ = api;
    } else {
      delete window.__REACT_GRAB__;
    }
  }
};
if (typeof window !== "undefined") {
  if (window.__REACT_GRAB__) {
    globalApi = window.__REACT_GRAB__;
  } else {
    globalApi = chunk6ZPKD3CA_cjs.init();
    window.__REACT_GRAB__ = globalApi;
    window.dispatchEvent(
      new CustomEvent("react-grab:init", { detail: globalApi })
    );
  }
}

Object.defineProperty(exports, "DEFAULT_THEME", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.DEFAULT_THEME; }
});
Object.defineProperty(exports, "formatElementInfo", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.getElementContext; }
});
Object.defineProperty(exports, "generateSnippet", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.generateSnippet; }
});
Object.defineProperty(exports, "getStack", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.getStack; }
});
Object.defineProperty(exports, "init", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.init; }
});
Object.defineProperty(exports, "isInstrumentationActive", {
  enumerable: true,
  get: function () { return chunk6ZPKD3CA_cjs.Ee; }
});
exports.getGlobalApi = getGlobalApi;
exports.setGlobalApi = setGlobalApi;
